src/openloco/windows/industry_list.cpp

```diff
@@ -1,16 +1,1272 @@
+            call(0x00458C09, regs);
+        }
+
+        // 0x00468FFE hide_gridlines
+        static void hideGridlines()
+        {
+            _gridlines_state--;
+            if (_gridlines_state == 0)
+            {
+                auto window = WindowManager::getMainWindow();
+                if (window != nullptr)
+                {
+                    if ((config::get().flags & config::flags::gridlines_on_landscape) == 0)
+                    {
+                        if ((window->viewports[0]->flags & viewport_flags::gridlines_on_landscape) == 0)
+                        {
+                            window->invalidate();
+                        }
+                        window->viewports[0]->flags ^= viewport_flags::gridlines_on_landscape;
+                    }
+                }
+            }
+        }
+
+        // 0x004585AD
+        static void on_tool_abort(window& self, const widget_index widgetIndex)
+        {
+            sub_458C09();
+            hideGridlines();
+        }
+
+        // 0x0045845F
+        static void on_close(window* self)
+        {
+            if (input::is_tool_active(self->type, self->number))
+                input::cancel_tool();
+        }
+
+        // 0x00458B51
+        static void updateActiveThumb(window* self)
+        {
+            uint16_t scrollHeight = 0;
+            self->call_get_scroll_size(0, 0, &scrollHeight);
+            self->scroll_areas[0].v_bottom = scrollHeight;
+
+            auto i = 0;
+            if (i >= self->var_83C)
+            {
+                for (; i <= self->var_83C; i++)
+                {
+                    if (self->row_info[i] == self->row_hover)
+                        break;
+                }
+            }
+            self->scroll_areas[0].v_top = i;
+            ui::scrollview::update_thumbs(self, new_industries::widx::scrollview);
+        }
+
+        // 0x00458AAF
+        static void updateBuildableIndustries(window* self)
+        {
+            auto industryCount = 0;
+            for (auto i = 0; objectmgr::get_max_objects(object_type::industry); i++)
+            {
+                auto industryObj = objectmgr::get<industry_object>(i);
+                if (industryObj == nullptr)
+                    break;
+                if (!is_editor_mode())
+                {
+                    if (!(industryObj->flags & industry_object_flags::can_be_founded_by_user))
+                        continue;
+                    if (current_year() < industryObj->designedYear)
+                        continue;
+                    if (current_year() > industryObj->obsoleteYear)
+                        continue;
+                }
+                self->row_info[industryCount] = i;
+                industryCount++;
+            }
+
+            self->var_83C = industryCount;
+            auto rowHover = -1;
+
+            if (byte_525FC7 != 0xFF)
+            {
+                for (auto i = 0; i <= self->var_83C; i++)
```

Comment:
This section is not correct. We are iterating over all of the available rows and checking to see if they are the same as 525FC7 if they are the same we rowHover = byte_525FC7. I would rewrite it as follows:

```
if (byte_525FC7 != 0xFF)
{
    for(auto i = 0; i < self->var_83C; i++)
    {
         if (byte_525FC7 == self->row_info[i])
         {
               rowHover = byte_525FC7;
               break;
         }
     }
}

if (rowHover == -1 && self->var_83C != 0)
{
    rowHover = self->row_info[0];
}

self->row_hover = rowHover;
updateActiveThumb(self);